sdk/core/azure-core/src/http/policy.cpp

```diff
@@ -16,16 +16,15 @@ Azure::Core::Logging::LogClassification const
     Azure::Core::Http::LogClassification::HttpTransportAdapter;
 #endif
 
+// The NextHttpPolicy can't be created from a nullptr because it is a reference. So we don't need to
+// check if m_policies is nullptr.
```

Comment:
Good comment. I am still a bit confused about the ref vs ptr semantics of C++ here. @antkmsft - can you please help clarify why this works?